File: dailyQuestion/2020-06/06-03/golang/solution_expandAroundCenter.go
package main

import (
	"fmt"
)

func main() {

	s := "abcabcbb"
	result := countSubstrings(s)
	fmt.Println(result)

}

func countSubstrings(s string) int {
	result := 0
	if s == "" {
		return result
	}
	for i := 0; i < len(s); i++ {
		result += expandAroundCentor(s, i, i)
		result += expandAroundCentor(s, i, i+1)
	}
	return result
}

func expandAroundCentor(s string, i, j int) int {
	result := 0
	for ; i >= 0 && j < len(s) && s[i] == s[j]; i, j = i-1, j+1 {
		result++
	}
	return result
}

File: dailyQuestion/2020/2020-06/06-03/python/solution_expandaroundcenter.py
def countSubstrings(s: str) -> int:
    def expendAroundCentor(s, i, j):
        result = 0
        while i >= 0 and j < len(s) and s[i] == s[j]:
            result += 1
            i -= 1
            j += 1

        return result

    result = 0
    if not s:
        return result
    for i in range(len(s)):
        result += expendAroundCentor(s, i, i)
        result += expendAroundCentor(s, i, i + 1)

    return result


if __name__ == "__main__":
    s = "aaa"
    result = countSubstrings(s)
    print(result)
